homeassistant/components/mysensors/config_flow.py

```diff
@@ -0,0 +1,224 @@
+        errors = {}
+        if user_input is not None:
+            if CONF_TCP_PORT in user_input:
+                port: int = user_input[CONF_TCP_PORT]
+                if not (0 < port <= 65535):
+                    errors[CONF_TCP_PORT] = "port_out_of_range"
+
+            errors.update(
+                await self.validate_common(CONF_GATEWAY_TYPE_TCP, errors, user_input)
+            )
+            if not errors:
+                return self.async_create_entry(
+                    title=f"{user_input[CONF_DEVICE]}", data=user_input
+                )
+
+        schema = _get_schema_common()
+        schema[vol.Required(CONF_DEVICE, default="127.0.0.1")] = str
+        # Don't use cv.port as that would show a slider *facepalm*
+        schema[vol.Optional(CONF_TCP_PORT, default=DEFAULT_TCP_PORT)] = vol.Coerce(int)
+
+        schema = vol.Schema(schema)
+        return self.async_show_form(step_id="gw_tcp", data_schema=schema, errors=errors)
+
+    async def async_step_gw_mqtt(self, user_input: Optional[Dict[str, str]] = None):
+        """Create a config entry for a mqtt gateway."""
+        errors = {}
+        if user_input is not None:
+            user_input[CONF_DEVICE] = MQTT_COMPONENT
+
+            if CONF_TOPIC_IN_PREFIX in user_input:
+                try:
+                    valid_subscribe_topic(user_input[CONF_TOPIC_IN_PREFIX])
+                except vol.Invalid:
+                    errors[CONF_TOPIC_IN_PREFIX] = "invalid_subscribe_topic"
+
+            if CONF_TOPIC_OUT_PREFIX in user_input:
+                try:
+                    valid_publish_topic(user_input[CONF_TOPIC_OUT_PREFIX])
+                except vol.Invalid:
+                    errors[CONF_TOPIC_OUT_PREFIX] = "invalid_publish_topic"
+
+            errors.update(
+                await self.validate_common(CONF_GATEWAY_TYPE_MQTT, errors, user_input)
+            )
+            if not errors:
+                return self.async_create_entry(
+                    title=f"{user_input[CONF_DEVICE]}", data=user_input
+                )
+        schema = _get_schema_common()
+        schema[vol.Required(CONF_RETAIN, default=True)] = bool
+        schema[vol.Optional(CONF_TOPIC_IN_PREFIX)] = str
+        schema[vol.Optional(CONF_TOPIC_OUT_PREFIX)] = str
+
+        schema = vol.Schema(schema)
+        _LOGGER.debug("mysconfigflow mqtt errors: %s", errors)
+        return self.async_show_form(
+            step_id="gw_mqtt", data_schema=schema, errors=errors
+        )
+
+    async def validate_common(
+        self,
+        gw_type: ConfGatewayType,
+        errors: Dict[str, str],
+        user_input: Optional[Dict[str, str]] = None,
+    ) -> Dict[str, str]:
+        """Validate parameters common to all gateway types."""
+        if user_input is not None:
+            errors.update(_validate_version(user_input.get(CONF_VERSION)))
+
+            if gw_type != CONF_GATEWAY_TYPE_MQTT:
+                verification_func = (
+                    is_socket_address
```

Comment:
We prefer to not use a ternary operator expression if the statement spans more than one line.